functions/nodejs-typescript-esbuild/src/libs/scalewayServerless.ts

```diff
@@ -0,0 +1,172 @@
+type SwcRegion = "fr-par" | "nl-ams" | "pl-waw";
+
+type Runtime =
+  | "node20"
+  | "node18"
+  | "node17"
+  | "node16"
+  | "node14"
+  | "node10"
+  | "node8"
+  | "python311"
+  | "python310"
+  | "python39"
+  | "python38"
+  | "python37"
+  | "python"
+  | "python3"
+  | "php82"
+  | "go120"
+  | "go119"
+  | "go118"
+  | "go117"
+  | "go113"
+  | "golang"
+  | "rust165";
+
+type MemoryLimit = 128 | 256 | 512 | 1024 | 2048 | 3072 | 4096;
+
+type ConfigValidationMode = "warn" | "error" | "off";
+
+type Events = any; // needs to be typed!
+
+interface Package {
+  individually?: false; // Not supported by scaleway
+  patterns?: Array<string>;
+}
+
+interface ScalewayServerlessProvider {
+  name: "scaleway";
+  runtime: Runtime;
+  env?: Record<string, string>;
+  secret?: Record<string, string>;
+
+  swcRegion: SwcRegion;
+}
+
+interface ScalewayFunctionTimeOut {
+  seconds: number;
+}
+type ScalewayFunctionScales =
+  | 0
+  | 1
+  | 2
+  | 3
+  | 4
+  | 5
+  | 6
+  | 7
+  | 8
+  | 9
+  | 10
+  | 11
+  | 12
+  | 13
+  | 14
+  | 15
+  | 16
+  | 17
+  | 18
+  | 19
+  | 20;
+export interface ScalewayFunction {
+  handler: string;
+  description?: string;
+  env?: Record<string, string>;
+  secret?: Record<string, string>;
+  minScale?: ScalewayFunctionScales;
+  maxScale?: ScalewayFunctionScales;
+  maxConcurrency?: number;
+  memoryLimit?: MemoryLimit;
+  // timeout?: any; NOT WORKING! 
```

Comment:
Will look into it, thanks!

Reply:
Im guessing timeout should be a number right? or are they set values ?